### PR TITLE
DOCS: Expand content/loaders 

### PR DESCRIPTION
**Pull request by mikeerickson**
Extended `content/concepts/loaders` section

- [x] Expand documentation, providing additional instructions
- [x] Added examples to accompany expanded points
- [x] Added `mikeerickson` to list of contributors (also updated output section)

@TheLarkInn Please review and let me know if you like the format used for the `loaders` section.  I basically expanded on what you have done elsewhere so it should flow the same.

### CODE REVIEW

**Comment by simon04** [comment]
This duplicates content from https://webpack.js.org/configuration/module/. Do we want to go into detail in the "concept" section? IMO, also https://webpack.js.org/concepts/output/ is too long/detailed for the "concept" section.

**Comment by bebraw** [comment]
Agreed. The content is valid, but it can be laid out better. It's enough if the concept article can describe the basic idea and link to other articles that dig into greater detail. You can have `## References` with links at the end.

**Comment by mikeerickson** [comment]
@bebraw @simon04 my approach (and you are free to say otherwise) is to document the minimum examples to get started.  

I am teaching a new JR to start using webpack and was met with friction as soon as getting to concepts/loaders as there was literally not even simple examples 

It's your call, but I still believe examples where users are getting started

**Comment by bebraw** [comment]
Have you seen https://webpack.js.org/loaders/ ? Probably the main thing missing is a reference between the concept page and that one. The concept page could contain a very basic example (the first one you added) and then defer to that article for more.

**Comment by mikeerickson** [comment]
@bebraw no, I didn't even know about  this section until I read this initial reply.  I will review this section and revise my edits and provide link accordingly

**Comment by bebraw** [comment]
@mikeerickson Ok, cool.

**Comment by mikeerickson** [comment]
@bebraw One quick question before I proceed.   In review of this page example (https://webpack.js.org/loaders/) it uses the `use` long syntax.   Is this the "preferred" method, or is it safe to use the shorter approach as documented in my edits?

```
  module: {
    rules: [
      {
        test: /\.css$/,
        use: [
          { loader: 'style-loader'},
          {
            loader: 'css-loader',
            options: {
              modules: true
            }
          }
        ]
      }
    ]
  }
```

vs 

```
module: {
  rules: [
    {test: /\.js$/, loader: 'babel-loader'},
    ... 
  ]
}
```

**Comment by bebraw** [comment]
@mikeerickson Both forms have their use I think. `use` is handy if you want to pass multiple loaders within a single definition. I have written a [little chapter](http://survivejs.com/webpack/understanding-loaders/loader-definitions/) with the key ideas.

**Comment by mikeerickson** [comment]
@bebraw Would it be prudent and example worthy, to demonstrate the `short` syntax in the loaders section (along with the `use` format)?   Or, just link to it from `concepts/loaders` and be done with it.   Just trying to make it as clear for beginners as possible.

**Comment by mikeerickson** [comment]
@bebraw In short, in the `Via Configuration` section, provide both short and extended examples, since this is what is used in the `concepts/loaders` section

**Comment by bebraw** [comment]
@mikeerickson I would go with a mini example and a "call to action" to encourage to read more from the in-depth article that digs into further details.

**Comment by mikeerickson** [comment]
@bebraw cool, that is the approach I have just applied  :-)

**Comment by mikeerickson** [comment]
@bebraw I have updated my PR, please have a look and let me know what you think.

File: content/concepts/loaders.md [hunk 1]
```diff
@@ -32,6 +33,64 @@ Loaders are [resolved similar to modules](/concepts/module-resolution/). A loade
...
+    rules: [
+      {test: /\.js$/, loader: 'babel-loader', exclude: /node_modules/}, // transform js files
+      {test: /\s[a|c]ss$/, loader: 'sass-loader'}                       // transform sass files
+      {test: /\.css$/, loaders: ['style-loader','css-loader']}          // transform css files
+    ]
+  }
+  ...
+};
+```
+#### Install Webpack Loader modules
```
Comment (bebraw) [inline content/concepts/loaders.md]: Can you check the header hierachy for both? I think you need ##/### at tops.

**Comment by mikeerickson** [comment]
@bebraw Whoops, a product of a copy/paste error.  All fixed :-)

File: content/concepts/loaders.md [hunk 2]
```diff
@@ -32,6 +33,64 @@ Loaders are [resolved similar to modules](/concepts/module-resolution/). A loade
...
+## Options
+
+The above example demonstrates the different approaches which can be used when defining the `module.rules` section of your `webpack.config.js` file.
+For complete details, refer to [loaders section](/loaders).
+
+### Omitting Loader Error
+
+If you fail to define a loader for the desired files within your project, you will receive an error similar to the following:
+
+`Module not found: Error: Can't resolve 'json-loader' in ...`
```
Comment (bebraw) [inline content/concepts/loaders.md]: json-loader is integrated to webpack 2. It would be a good idea to replace this example with some other format.

Comment (mikeerickson) [inline content/concepts/loaders.md]: @bebraw ah, I didnt know that :-)   It was already used as a basic example, so I just kept it around.  I will adjust to continue using the css stuff I have demonstrated (or maybe sass-loader)

File: content/concepts/loaders.md [hunk 3]
```diff
@@ -32,6 +33,64 @@ Loaders are [resolved similar to modules](/concepts/module-resolution/). A loade
...
+
+**webpack.config.js**
+
+```javascript
+const config = {
+  ...
+  module: {
+    rules: [
+      {test: /\.js$/, loader: 'babel-loader', exclude: /node_modules/}, // transform js files
+      {test: /\s[a|c]ss$/, loader: 'sass-loader'}                       // transform sass files
```
Comment (bebraw) [inline content/concepts/loaders.md]: Why before the other styling loader? Shouldn't sass bits get evaluated first (evaluation order for loaders is from bottom to top if I remember right).

**Comment by bebraw** [comment]
@simon04 Do you want to have a look? I could use a second opinion here.

**Comment by simon04** [comment]
I'm not convinced:
I would rather take the _simple_, textual example from the intro paragraph ("you can use loaders to tell webpack to load a CSS file or to convert TypeScript to JavaScript") and develop a code example from there. Done in #736.
Also, it's good to know that there are various options to specify loaders, but I would just quickly sketch them and refer to the configuration page. Done in #736.

**Comment by mikeerickson** [comment]
@bebraw @simon04 would you like me to adjust my changes, or just let it go (and accept #736)?  I am keen to do as you wish, just tell me what you want (since you are not convinced by my contribution).

**Comment by bebraw** [comment]
@mikeerickson I think we'll accept #736 first and then figure out how to revise. I'll get back to you after that.

**Comment by bebraw** [comment]
If you want, you could rebase it so that the PR retains `Omitting Loader Error` and `Webpack Rules Shorthand Syntax` sections. There's a typo in the latter (sytnax) to fix.

**Pull request closed without merging**